superset/connectors/druid/models.py

```diff
@@ -817,7 +829,7 @@ def granularity(
             "year": "P1Y",
         }
 
-        granularity: Dict[str, Union[str, float]] = {"type": "period"}
+        granularity = {"type": "period"}
```

Comment:
I would have thought this turns into a `Dict[str, str]` unless being explicitly being defined as `Granularity`, and thus causing problems below when numeric values are added?

Reply:
@villebro I guess it differs from the `Granularity` type which I defined previously (which adds to the confusion). When including the typing here it causes a conflict with the return type as it can also be a string per line 812. 

A couple of learnings from having typed a bunch of Python code: 

1. `Optional[...]` values are difficult to handle and thus should be avoided if possible. It seems they're often used by laziness. 
2. Mixed types, i.e., those requiring a `Union[...]` normally indicates poor quality and unnecessarily complex code. 

I think the return type of `granularity` is a clear indication of (2). Also I believe the true return type should be `Union[Dict[str, Union[str, float]], str]` but `mypy` complains on this.